### PR TITLE
feat(sonarqube): add recipe

### PR DESCRIPTION
**Pull request by YaMoef**
#### Pre-flight Checklist

Please ensure you've completed all of the following.

- [x] I have read the [Contributing Guidelines](https://github.com/ferdium/ferdium-app/blob/develop/CONTRIBUTING.md) for this project.
- [x] I agree to follow the [Code of Conduct](https://github.com/ferdium/ferdium-app/blob/develop/CODE_OF_CONDUCT.md) that this project adheres to.
- [x] I updated the version package [Updating](https://github.com/ferdium/ferdium-recipes/blob/main/docs/updating.md#4-updating-the-version-number) 

#### Description of Change

Added new recipe: SonarQube

### CODE REVIEW

File: recipes/sonarqube/package.json [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
+  "id": "sonarqube",
+  "name": "SonarQube",
+  "version": "1.0.1",
```
Comment (vraravam) [inline recipes/sonarqube/package.json:4]: please change the version for any new service recipe to start from `1.0.0`

File: recipes/sonarqube/index.js [hunk 2]
```diff
@@ -0,0 +1,6 @@
+module.exports = Ferdium =>
+  class SonarQube extends Ferdium {
+    buildUrl(url) {
+      return `${url}/`;
```
Comment (vraravam) [inline recipes/sonarqube/index.js:4]: should this be defensive and first check if the url already ends with a `/`?

**Review by vraravam — CHANGES_REQUESTED** [review]

File: recipes/sonarqube/index.js [hunk 3]
```diff
@@ -1,6 +1,6 @@
 module.exports = Ferdium =>
   class SonarQube extends Ferdium {
     buildUrl(url) {
-      return `${url}/`;
+      return `${url}`;
```
Comment (vraravam) [inline recipes/sonarqube/index.js:4]: What's the point of defining this function? Please remove.

**Review by vraravam — CHANGES_REQUESTED** [review]

File: recipes/sonarqube/index.js (hunk 3 above)
Comment (YaMoef) [reply recipes/sonarqube/index.js]: Fixed on other PR as well, excuse me for the delay :)

**Review by vraravam — APPROVED** [review]
haven't tested, but looks good to me

**Review by vraravam — CHANGES_REQUESTED** [review]
@YaMoef  - looks like the CI run is failing. Most likely due to formatting changes that show up as modified files. Please run the commands locally and fix them before this PR can be approved and merged.

**Comment by YaMoef** [comment]
> @YaMoef - looks like the CI run is failing. Most likely due to formatting changes that show up as modified files. Please run the commands locally and fix them before this PR can be approved and merged.

I ran the formatting commands, but I get locally issues with a version bump is that ok? Or do I need to bump the version?

**Comment by vraravam** [comment]
> I ran the formatting commands, but I get locally issues with a version bump, is that ok? Or do I need to bump the version?

you can run the git commit command by bypassing the precommit checks with the `-n` switch. with this you can bypass the check for version bump.

**Comment by YaMoef** [comment]
> > I ran the formatting commands, but I get locally issues with a version bump, is that ok? Or do I need to bump the version?
> 
> you can run the git commit command by bypassing the precommit checks with the `-n` switch. with this you can bypass the check for version bump.

Perfect, this is what I did when I committed. The PR is good to go on my end :)

**Comment by vraravam** [comment]
@YaMoef  - it is always a good idea to rebase from the latest version of the parent branch (`main` in this case) so that all of us can be assured that your changes will pass the CI tests *on top of* the latest so that when merged, the build will remain green. I have rebased now, but in the future (at least if contributing to Ferdium's repos), please ensure that your PR is rebased.

**Review by vraravam — APPROVED** [review]
LGTM

**Pull request merged**